### PR TITLE
[Merged by Bors] - feat: simp lemmas for polynomial coefficients

### PR DESCRIPTION
**Pull request by PatrickMassot**
Code by Ruben-VandeVelde

Co-authored-by: Ruben Van de Velde <65514131+Ruben-VandeVelde@users.noreply.github.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Basic.lean [hunk 1]
```diff
@@ -757,6 +757,22 @@ lemma coeff_C_succ {r : R} {n : ℕ} : coeff (C r) (n + 1) = 0 := by simp [coeff
 theorem coeff_nat_cast_ite : (Nat.cast m : R[X]).coeff n = ite (n = 0) m 0 := by
   simp only [← C_eq_nat_cast, coeff_C, Nat.cast_ite, Nat.cast_zero]
 
+@[simp]
+theorem coeff_natCast_zero (a : ℕ) : coeff a 0 = a := coeff_monomial
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/Basic.lean:761]: Doesn't this `a` need a type ascription?

Comment (eric-wieser) [reply Mathlib/Data/Polynomial/Basic.lean:761]: An explanation of `no_index` would be good here.

**Comment by eric-wieser** [comment]
bors d+

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**